portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -196,6 +198,14 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
      */
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // get reference to UserService
+        UserService userService = UserServiceFactory.getUserService();
+        // if user is not logged in, s/he can't post comments - send her/him to "/"
```

Comment:
I found 'they', 'them' more useful to refer to a user and easier to read. But it's up to you.